### ISSUE TITLE
Dialogs should capture keyboard focus

### ISSUE THREAD

**Issue by PRGfx** [neos/neos-ui#3397]
### Description

When a dialog is open, it should not be possible to tab out of the dialog. The focus should return to the first (or last) element in the dialog.
When the dialog is always focused, that would also help with consistent escape-closing

### Steps to Reproduce

1. (probably add some global styling so we actually see the focused element.. `*:focus-visible{outline:2px solid red !important}`)
2. Open a dialog, for example the node-creation dialog
3. (Shift-)Tab around and observer that everything can be focused

#### Expected behavior

Opening the dialog should focus an appropriate element within the dialog and tabbing forwards or backwards should not leave the dialog.
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/#keyboardinteraction

There might be cases when opening the dialog doesn't need to focus the first element (see the publish button in #3032).

#### Actual behavior

As the dialog is portaled to the end of the document, tabbing forwards goes through the browser controls and to the guest-frame, tabbing backwards goes back into the Neos UI.

### Affected Versions

probably every version

**Comment by minedesign-jl** [comment]
I propose switching to the native `<dialog>` element for proper tabbing order and focus management.

For this switch issues like #3582 and #3679 should be kept in mind to avoid losing progress (e. g. in node creation) when escaping the dialog. Afaik a decision about how to best handle the case "warning before closing" within a dialog hasn't been made.
